polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -242,17 +241,26 @@ unsigned getLocalMemoryRemaining(gpu::GPUModuleOp &module,
   return localMemoryRemaining;
 }
 
-/// Get the required local memory for \p accTy in bytes.
+/// Get the amount of shared memory needed by \p accTy in bytes.
 std::variant<Value, unsigned>
 getReqdLocalMemory(sycl::AccessorType accTy, const WorkGroupSize &workGroupSize,
                    OpBuilder builder) {
   unsigned elemSize = accTy.getType().getIntOrFloatBitWidth() / 8;
-  std::variant<Value, unsigned> reqdLocalMemory =
-      ValueOrUnsigned::get(elemSize, builder, workGroupSize.hasElemTy<Value>());
   const unsigned numDims = accTy.getDimension();
+
+  if (workGroupSize.isKnown()) {
+    unsigned memrefReqdLocalMemory = elemSize;
+    for (unsigned dim = 0; dim < numDims; ++dim)
+      memrefReqdLocalMemory *= workGroupSize.get<unsigned>(dim);
+    return memrefReqdLocalMemory;
+  }
+
```

Comment:
can probably be removed.

Reply:
Yes I kind of merged/simplified the original code and forgot that `ValueOrUnsigned::[get|mul]` return an variant containing an unsigned when given unsigned arguments.  I'll remove these section of code. Thanks.